modules/boltlib/lib/puppet/functions/run_task.rb

```diff
@@ -10,42 +10,29 @@
     type 'TargetOrTargets = Variant[String[1], Target, Array[TargetOrTargets]]'
   end
 
-  dispatch :run_task_type do
-    param 'Type[Task]', :task_type
-    param 'TargetOrTargets', :targets
-    optional_param 'Hash[String[1], Any]', :task_args
-  end
-
   dispatch :run_named_task do
-    param 'String[1]', :task_type
+    param 'String[1]', :task_name
     param 'TargetOrTargets', :targets
     optional_param 'Hash[String[1], Any]', :task_args
   end
 
-  dispatch :run_task_instance do
+  dispatch :run_task do
     param 'Task', :task
-    repeated_param 'TargetOrTargets', :targets
-  end
-
-  def run_task_type(task_type, targets, task_args = nil)
-    use_args = task_args.nil? ? {} : task_args
-    task_instance = call_function('new', task_type, use_args)
-    run_task_instance(task_instance, targets)
+    param 'TargetOrTargets', :targets
+    optional_param 'Hash[String[1], Any]', :task_args
   end
```

Comment:
This commit reduces the `run_task` function signatures to:
```ruby
dispatch :run_named_task do
   param 'String[1]', :task_name
   param 'TargetOrTargets', :targets
   optional_param 'Hash[String[1], Any]', :task_args
end

dispatch :run_named_task do
   param 'Task', :task
   param 'TargetOrTargets', :targets
   optional_param 'Hash[String[1], Any]', :task_args
end
```
Does it make sense to provide the vararg signature shown below?
```ruby
dispatch :run_task_vararg do
   param 'Task', :task
   repeated_param 'TargetOrTargets', :targets
end
```
Note that this signature could only be used for running tasks which don't require any arguments.